File: AtlasWeb/Core/wwwroot/app/NebulaClient/Visual/ComboListItem.ts
export class ComboListItem {

    private _dataValue: any;
    public get DataValue(): any {
        return this._dataValue;
    }

    private _displayText: string;
    public get DisplayText(): string {
        return this._displayText;
    }

    public constructor(dataValue: any, displayText: string) {
        this._displayText = displayText;
        this._dataValue = dataValue;
    }
}

File: AtlasWeb/Infrastructure/Models/ComboListItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Infrastructure.Models
{
    public class ComboListItem
    {
        private object _dataValue;
        public object DataValue
        {
            get
            {
                return _dataValue;
            }
        }

        private string _displayText;
        public string DisplayText
        {
            get
            {
                return _displayText;
            }
        }

        public ComboListItem(object dataValue, string displayText)
        {
            _displayText = displayText;
            _dataValue = dataValue;
        }
    }
}